sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2185,7 +2185,11 @@ class Analyzer(override val catalogManager: CatalogManager)
               externalFunctionNameSet.add(fullName)
               f
             } else {
-              throw QueryCompilationErrors.noSuchFunctionError(nameParts, f, Some(fullName))
+              val catalogPath = (catalog.name() +: catalog.defaultNamespace()).mkString(".")
```

Comment:
I think we should use the current database, not the catalog default namespace. If you look at `CatalogAndIdentifier`, the resolution code here does check `catalogManager.currentNamespace`

Reply:
Thank you, will replace it by `catalogManager.currentNamespace`.